src/Uno.UI/UI/Xaml/UIElement.Properties.cs

```diff
@@ -68,6 +69,24 @@ Visibility Visibility
 			set => SetVisibilityValue(value);
 		}
 
+		/// <summary>
+		/// Represents the final cursor shape of the element that is set using ProtectedCursor.
+		/// </summary>
+		/// <remarks>
+		/// This property should never be directly set except from ProtectedCursor's ChangedCallback, and its value should always be calculated through the dependency property system.
+		/// </remarks>
+		/// <remarks>
+		/// The type is nullable because we need a state that indicates that the cursor should be hidden. We choose that state to be null.
+		/// </remarks>
+		[GeneratedDependencyProperty(DefaultValue = InputSystemCursorShape.Arrow, Options = FrameworkPropertyMetadataOptions.Inherits, ChangedCallback = false)]
+		internal static DependencyProperty CalculatedFinalCursorProperty { get; } = CreateCalculatedFinalCursorProperty();
```

Comment:
Normally we use `GeneratedDependencyProperty` for DPs that are frequently read. I'd personally lean towards a regular DP declaration for this case